src/main/haskell/kore/test/Test/Kore/Builtin/List.hs

```diff
@@ -0,0 +1,275 @@
+module Test.Kore.Builtin.List where
+
+import Test.QuickCheck
+       ( Property, property, (.&&.), (===) )
+
+import           Data.Map
+                 ( Map )
+import qualified Data.Map as Map
+import           Data.Proxy
+                 ( Proxy (..) )
+import           Data.Reflection
+                 ( give )
+import           Data.Sequence
+                 ( Seq (..) )
+import qualified Data.Sequence as Seq
+
+import           Kore.AST.Common
+import           Kore.AST.MetaOrObject
+                 ( Object )
+import           Kore.AST.PureML
+                 ( CommonPurePattern )
+import           Kore.AST.Sentence
+import           Kore.ASTUtils.SmartConstructors
+import           Kore.ASTUtils.SmartPatterns
+import           Kore.ASTVerifier.DefinitionVerifier
+import           Kore.Attribute.Parser
+                 ( ParseAttributes (..) )
+import qualified Kore.Builtin as Builtin
+import           Kore.Builtin.Hook
+                 ( hookAttribute )
+import qualified Kore.Builtin.List as List
+import qualified Kore.Error
+import           Kore.IndexedModule.IndexedModule
+import           Kore.IndexedModule.MetadataTools
+import           Kore.Step.ExpandedPattern
+import qualified Kore.Step.ExpandedPattern as ExpandedPattern
+import           Kore.Step.Simplification.Data
+import qualified Kore.Step.Simplification.Pattern as Pattern
+import           Kore.Step.StepperAttributes
+                 ( StepperAttributes )
+
+import           Test.Kore
+                 ( testId )
+import qualified Test.Kore.Builtin.Bool as Test.Bool
+import           Test.Kore.Builtin.Builtin
+import qualified Test.Kore.Builtin.Int as Test.Int
+
+{- |
+    @
+        get{}(unit{}(), _) === \bottom{}()
+    @
+ -}
+prop_getUnit :: Integer -> Property
+prop_getUnit k =
+    let patGet = App_ symbolGet [App_ symbolUnit [], Test.Int.asPattern k]
+        predicate = give testSortTools $ mkEquals mkBottom patGet
+    in
+        allProperties
+            [ ExpandedPattern.bottom === evaluate patGet
+            , ExpandedPattern.top === evaluate predicate
+            ]
+
+{- |
+    @
+        get{}(concat{}(element{}(e), ...), 0) === e
+    @
+ -}
+prop_getFirstElement :: [Integer] -> Property
+prop_getFirstElement (Seq.fromList -> values) =
```

Comment:
What does `Seq.fromList -> values` construct mean?

Reply:
That is a [view pattern](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#view-patterns); it applies `Seq.fromList` before binding `values`, so that `values :: Seq Integer` even though the first argument to `prop_getFirstElement` has type `[Integer]`. I should change the function signature.